src/clr-angular/utils/focus/key-focus/key-focus.ts

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+  AfterContentChecked,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Inject,
+  Input,
+  Output,
+  OnInit,
+  PLATFORM_ID,
+} from '@angular/core';
+
+import { isPlatformBrowser } from '@angular/common';
+
+import { Observable } from 'rxjs';
+
+@Directive({
+  selector: '[clrKeyFocus]',
+})
+export class ClrKeyFocus implements OnInit, AfterContentChecked {
+  constructor(private elementRef: ElementRef, @Inject(PLATFORM_ID) private platformId: Object) {}
+
+  @Input('clrKeyFocus') direction: 'horizontal' | 'vertical' | 'both';
+  @Input('clrFocusOnLoad') focusOnLoad: boolean = false;
+
+  private _focusChange: EventEmitter<void> = new EventEmitter();
+
+  @Output('clrFocusChanged')
+  get focusChange(): Observable<void> {
+    return this._focusChange.asObservable();
+  }
+
+  private _current: number = 0;
+  get current() {
+    return this._current;
+  }
+
+  ngOnInit() {
+    if (!this.direction) {
+      this.direction = 'vertical';
+    }
+    if (this.focusOnLoad && isPlatformBrowser(this.platformId)) {
+      this.getCurrent().focus();
+      this._focusChange.next();
+    }
+  }
+
+  ngAfterContentChecked() {
+    this.getChildren().forEach(button => {
+      this.setTabIndex(button, -1);
+    });
+    this.setTabIndex(this.getCurrent(), 0);
+  }
+
+  @HostListener('keydown', ['$event'])
+  handleKeyboardEvent(event: any) {
+    if (this.prevClicked(event) && this._current - 1 >= 0) {
+      this.keyAction(() => {
+        this._current--;
+      });
+    } else if (this.nextClicked(event) && this._current + 1 < this.getChildren().length) {
```

Comment:
nit, do you mind leaving an inline comment to explain these checks in plain language?

Reply:
sure!

Reply:
done